File: pow.go
package main
import "fmt"

func pow(a int, b int) int {
	for r := 1; true; {
    if b <= 0 { return r }
    b--
		r *= a
	}
  return 1
}

func main() {
    fmt.Println(pow(2, 4))
}

File: pow.java
class Math {
  public static int pow(int a, int b) {
    if (b <= 0)
      return 1;
    return pow(a, b - 1) * a;
  }
}

class Main {  
  public static void main(String args[]) {
    System.out.println(Math.pow(2, 4)); 
  }
}
